### ISSUE TITLE
macOS: "Secure coding is automatically enabled for restorable state!" warning message

### ISSUE THREAD

**Issue by agarny** [opencor/opencor#2748]
If we run OpenCOR from the command line on macOS 14 (Sonoma) then we get the following warning message:
```
2024-01-30 23:21:58.371 OpenCOR[35689:2002971] WARNING: Secure coding is automatically enabled for restorable state! However, not on all supported macOS versions of this application. Opt-in to secure coding explicitly by implementing NSApplicationDelegate.applicationSupportsSecureRestorableState:.
```
This is something that only affects macOS 12+ and that is implicitly enabled on macOS 14. See [here](https://sector7.computest.nl/post/2022-08-process-injection-breaking-all-macos-security-layers-with-a-single-vulnerability/) for more details.

**Comment by agarny** [comment]
Note that Qt overloads the NSApplicationDelegate class (QCocoaApplicationDelegate), so ideally we would overload Qt's interface, but... it's private, so we can't!

To ignore the message in [messageHandler()](https://github.com/opencor/opencor/blob/master/src/misc/cliutils.cpp#L59-L73) doesn't work either since it's a pure macOS warning, so not something that Qt can control.

The only option seems to be to upgrade to Qt 6 (see issue #2718) since it handles that macOS warning, but... we need Qt WebKit and it has yet to be ported to Qt 6. (In fact, I wouldn't expect Qt WebKit to ever be ported to Qt 6 since it is already lagging behind big time...)

**Comment by agarny** [comment]
Just to confirm that this issue has been fixed in Qt 6.2.11, 6.5.4, and 6.6.1, as well as the forthcoming Qt 6.7.0. In other words, it is in their LTS versions of Qt (i.e. Qt 6.2 and Qt 6.5).

**Comment by agarny** [comment]
We decided to stick to Qt 5.12.12 LTS, so we can't fix this issue hence closing.
